packages/dev/core/src/XR/features/WebXRLayers.ts

```diff
@@ -46,50 +46,57 @@ class WebXRCompositionLayerRenderTargetTextureProvider extends WebXRLayerRenderT
     protected _getRenderTargetForSubImage(subImage: XRWebGLSubImage, eye: XREye) {
         const lastSubImage = this._lastSubImages.get(eye);
         const eyeIndex = eye == "left" ? 0 : 1;
-        if (!this._renderTargetTextures[eyeIndex] || lastSubImage?.textureWidth !== subImage.textureWidth || lastSubImage?.textureHeight != subImage.textureHeight) {
+
+        const colorTextureWidth = subImage.colorTextureWidth ?? subImage.textureWidth;
+        const colorTextureHeight = subImage.colorTextureHeight ?? subImage.textureHeight;
+
+        if (!this._renderTargetTextures[eyeIndex] || lastSubImage?.textureWidth !== colorTextureWidth || lastSubImage?.textureHeight != colorTextureHeight) {

```

Comment:
nit: last comparison is using != instead of !==

Reply:
Done